src/engine/Join.cpp

```diff
@@ -819,13 +790,13 @@ ProtoResult Join::computeResultForIndexScanAndIdTable(
             !idTable.empty() && idTable.at(0, joinColTable).isUndefined();
         std::optional<std::shared_ptr<const Result>> indexScanResult =
             std::nullopt;
-        using FirstColView = ad_utility::IdTableAndFirstCol<IdTable>;
-        using GenWithDetails =
-            cppcoro::generator<FirstColView,
-                               CompressedRelationReader::LazyScanMetadata>;
+        using GenWithDetails = decltype(convertGenerator(
```

Comment:
It seems the `convertGenerator` function is overloaded, but I don't currently see it...